### PR TITLE
Adds support for 3 gasses to the SM

### PR DESCRIPTION
**Pull request by redmoogle**
## About The Pull Request

Adds support for 3 new gasses to the Supermatter

- Tritium which has a heat penalty of 10(Plasma is 15) as strong as plasma but makes the Supermatter emit up-to 3x radiation at 100%

- Pluoxium is a weaker form of n2o. It makes the SM produce less heat at -1(Nitrogen has -1.5), Only works in concentrations above 15%, Reduces radiation by 2x at 100%(2500 compared to 7500), and has a heat buff of 3(N2O is 6)

- BZ starts emitting rad balls at concentrations above 40% and has at maximum of 30% chance of emitting one every tick, carries a heat penalty of 5(Plasma is 15) and also emits up-to 5x radiation at 100% (Also has a power transmission debuff to make the radiation not increase the power to insane levels)

**it's still dangerous to run engines with a heat penalty of 5; co2 has a heat penalty of 0.1**

## Why It's Good For The Game

- Adding more support for gasses to the Sm allows for different _and more dangerous_ setups
- Cleans the code up a tiny bit
- Adds some code to build off of for more advanced effects

## Changelog
:cl:
add: Added support for 3 new gasses; Tritium, Pluoxium, and BZ
/:cl:

### CODE REVIEW

**Comment by TheChosenEvilOne** [comment]
if I am being honest, only interaction I like is BZ.
Tritium is a quick delam gas and pluox is just plasma but less unstable.

**Comment by redmoogle** [comment]
> 
> 
> if I am being honest, only interaction I like is BZ.
> Tritium is a quick delam gas and pluox is just plasma but less unstable.

these were suggested and thats what i went with and to be honest I'd like more unique reactions as well

**Comment by TheChosenEvilOne** [comment]
so you just made tritium worse BZ that can cause a delamination?

**Comment by redmoogle** [comment]
> 
> 
> so you just made tritium worse BZ that can cause a delamination?

>So what Plasma can too and it works better

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -398,7 +408,10 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		power = max( (removed.temperature * temp_factor / T0C) * gasmix_power_ratio + power, 0) //Total laser power plus an overload
 
 		if(prob(50))
-			radiation_pulse(src, power * (1 + power_transmission_bonus/10))
+			radiation_pulse(src, power * (1 + (tritiumcomp*3)*(power_transmission_bonus/(10-(bzcomp*5)))))	// Emit upto 5x the radiation at 100% BZ & 3x Rad amount at 100% Tritum
+		if(prob(10 * (bzcomp * 3)) && bzcomp > 0.4)
```
Comment (Putnam3145) [inline code/modules/power/supermatter/supermatter.dm]: suuuuper minor micro optimization but short-circuiting means you should probably put the bzcomp check before the prob call

Comment (redmoogle) [reply code/modules/power/supermatter/supermatter.dm]: I wrote it to work not to be efficient though I'll keep it in mind

**Comment by granpawalton** [comment]
Pluxonium is known for being nonflammable and just generally more friendly in every aspect, why not make it so it makes the engine safer to match or have it cause some neutral effect?

File: code/modules/power/supermatter/supermatter.dm (hunk 1 above)
Comment (Putnam3145) [reply code/modules/power/supermatter/supermatter.dm]: this ain't criticism this is idle thoughts while reading code, i ain't a maintainer you can ignore me

Comment (redmoogle) [reply code/modules/power/supermatter/supermatter.dm]: I know

**Comment by redmoogle** [comment]
> 
> 
> Pluxonium is known for being nonflammable and just generally more friendly in every aspect, why not make it so it makes the engine safer to match or have it cause some neutral effect?

It was drawn from suggestions though having another sfatey gas would be nicer i guess

**Comment by Putnam3145** [comment]
that pluoxium change makes it WAY better than n2o, actually probably the best "safety gas" available, though n2o was kind of terrible as a safety gas since decomposition was added (i've seen people prefer bz over n2o just because at least n2o won't turn into oxygen and make your fires *worse*)

**Comment by redmoogle** [comment]
N2O would be more for mixing rather than emergency delam fuel also getting the % required is gonna be fun times

**Comment by Putnam3145** [comment]
for those who don't want to read the code: pluoxium has the heat resistance effect of N2O, but 1/6 as strong; it also has the heat penalty reduction of nitrogen, but 2/3 as strong; and it contributes to the power ratio the same way CO2, plasma, and oxygen do (also tritium, with these changes). This makes it a pretty effective safety gas that ALSO happens to help increase the power, the perfect gas to cut CO2 with for a high-power mix, basically a super-safe version of oxygen, which is perfectly in line with pluoxium's other roles as a gas but considering the ease of making it with the supermatter might be a bit much.

I actually like this change since pluoxium takes at least a LITTLE effort to make. My previous gas mix was just 3-10% n2o, 90-97% co2 and let that just sort of stew without the emitters, it's safe and easy. With this change I think one could be safe with a much bolder mix like 80% co2 20% oxygen, which will inevitably get all the oxygen replaced with pluox if you stop pumping the mix in, and it'd generate a lot of power. This seems balanced to me mostly because the basic roundstart N2 setup with emitters on is sufficient to power the entire station anyway, which is... really its own issue.

**Comment by redmoogle** [comment]
you got that slightly wrong it has 1/2 the heat resistance of n2o and doesn't help contribute(but doesn't directly hurt it) to power production and would be more viable for say a plasma setup

**Comment by Putnam3145** [comment]
thought i said "gasmix power ratio" but apparently i didn't upon rereading, whoops

**Comment by TetraK1** [comment]
>Pluxonium is known for being nonflammable and just generally more friendly in every aspect, why not make it so it makes the engine safer to match or have it cause some neutral effect?

I don't really think this reasoning is meaningful, and preferred the previous version.  Also, pluox being a safety gas makes co2 setups tend towards stable, which irks me, especially when co2 setups are way safer than their return implies they should be.  

BZ seems too strong when you can just steal the xenobio can.  Probably way too strong/griefy to balance it by making mass hallucinations (like after a delam) occur when it's in the loop, but would be neat as hell.

Tritium seems cool.

**Comment by FloranOtten** [comment]
BZ in Xenobio is never used. I can't be arsed to properly read code so having skimmed it, BZ seems overpowered. Brainless add can to SM for super power production at no risk. Why ever use tritium or plasma when BZ is available?

**Comment by redmoogle** [comment]
> 
> 
> BZ in Xenobio is never used. I can't be arsed to properly read code so having skimmed it, BZ seems overpowered. Brainless add can to SM for super power production at no risk. Why ever use tritium or plasma when BZ is available?

it makes the room more radioactive, not as much as I'd like, and shoots radballs I should probably attach a negative power transmission bonus

**Comment by redmoogle** [comment]
> 
> 
> > Pluxonium is known for being nonflammable and just generally more friendly in every aspect, why not make it so it makes the engine safer to match or have it cause some neutral effect?
> 
> I don't really think this reasoning is meaningful, and preferred the previous version. Also, pluox being a safety gas makes co2 setups tend towards stable, which irks me, especially when co2 setups are way safer than their return implies they should be.
> 
> BZ seems too strong when you can just steal the xenobio can. Probably way too strong/griefy to balance it by making mass hallucinations (like after a delam) occur when it's in the loop, but would be neat as hell.
> 
> Tritium seems cool.

I think having a 2nd 'Safety' gas would be better whilst also being weaker than n2o

**Comment by granpawalton** [comment]
I mean I don't really think that there is any good reasoning to make the only setup that isn't n2 and wont get you lynched closer to one that is, id rather it if it didn't have an effect or only had an effect after a certain partial pressure to keep the two types of engine separate

**Comment by redmoogle** [comment]
> 
> 
> I mean I don't really think that there is any good reasoning to make the only setup that isn't n2 and wont get you lynched closer to one that is, id rather it if it didn't have an effect or only had an effect after a certain partial pressure to keep the two types of engine separate

Making the engine do partial pressure stuff would involve adding most likely quite a bit more code as it can only see total moles and percents and wouldn't be a good idea to do math with it

**Comment by granpawalton** [comment]
I meant it in reference to the comment saying plux should be more dangerous and I was talking about how any CO2 setup is gonna be part plux setup basically which isn't necessary

I thought they had a partial pressure variable in sm code but the percent variable works fine for what I meant

IE. 
```java
if(plux percent > 10%) // or however much not filtering out for 10-20 mins is
   do something that doesnt make it closer or farther from deliming
   //the radiation particles was a good example
   //lowering radiation produced is one example
   //spawning something small in the surrounding area is another
```

If you don't want to handle what i'm talking about with plux don't worry about it

**Comment by redmoogle** [comment]
> 
> 
> I meant it in reference to the comment saying plux should be more dangerous and I was talking about how any CO2 setup is gonna be part plux setup basically which isn't necessary
> 
> I thought they had a partial pressure variable in sm code but the percent variable works fine for what I meant
> 
> IE.
> 
> ```java
> if(plux percent > 10%) // or however much not filtering out for 10-20 mins is
>    do something that doesnt make it closer or farther from deliming
>    //the radiation particles was a good example
>    //lowering radiation produced is one example
>    //spawning something small in the surrounding area is another
> ```
> 
> If you don't want to handle what i'm talking about with plux don't worry about it

ah ok I'll think on it

**Comment by redmoogle** [comment]
I'm gonna the negative rad modifier later today

**Comment by Cyprex** [comment]
Adding pluoxium to gasmix_power_ratio has implications for pluoxium production

**Comment by redmoogle** [comment]
Pluoxium rad reduction working as intended 2500(on pure pluox) Rads as compared to 7500(on pure n2) when directly next to the SM

**Comment by redmoogle** [comment]
> 
> 
> Adding pluoxium to gasmix_power_ratio has implications for pluoxium production

gasmix_power_ratio is basically how "high" the engines power can go

**Comment by FloranOtten** [comment]
>it's still dangerous to run engines with a heat penalty of 5; co2 has a heat penalty of 0.1

I would like to point out that the heat penalty isn't the danger in co2 engines, it's the stopping of power decay.

**Comment by redmoogle** [comment]
> > it's still dangerous to run engines with a heat penalty of 5; co2 has a heat penalty of 0.1
> 
> I would like to point out that the heat penalty isn't the danger in co2 engines, it's the stopping of power decay.

I know but just to keep the new engines that are dumb aware

**Comment by Cornarias** [comment]
I love this PR. I love everything about this PR. This PR makes me happy and I look forward to tinkering with it. Right now I've done all I can think of with the SM. Turn it into a burn chamber, self-sustained, single tile, sitting at power delam, etc. Most gasses means more customizability means more things to do, which generally means a lot more fun.

As for your comment about increasing the power delam point: YEESSSSS. I love pushing the SM to bordering on breaking point. The more I can push, the better.

**Pull request merged**